Simple Linear Regression
-----

**Question 1**  

The scatterplots below show the relationship between height, diameter, and volume of timber in 31 felled black cherry trees. The diameter of the tree is measured 4.5 feet above the ground.  

<img src="images/lab6_1_q4.png" width="600">  

(a) Describe the relationship between volume and height of these trees.  

There is a linear relationship between the height and Volume of timber. Samples are evenly distributed but there seam to be many outliers.

(b) Describe the relationship between volume and diameter of these trees.  

Strong linear correlation between diameter and tinder volume. Samples are a clustered in the smaler diameter range. There is one outlier with a diameter of 20 inches but he seems to fit the linear correlation.

(c) Suppose you have height and diameter measurements for another black cherry tree. Which of these variables would be preferable to use to predict the volume of timber in this tree using a simple linear regression model? Explain your reasoning.

Since the visual correlation between diameter and volume is stronger I would prefer to use it.

**Question 2**  

In `utils.csv` are the average utility bills for homes of a particular size and the average monthly temperature.  

Load the data:

In [1]:
import pandas as pd
df = pd.read_csv('./data/utils.csv')

In [2]:
df.head()

Month  Average Bill  Average Monthly Temperature
0   Jan         148.5                           38
1   Feb         136.5                           45
2   Mar         117.0                           49
3   Apr          91.5                           57
4   May          82.5                           69

Make a scatter plot of the data bill by tempeture. Which variable would make the most sense as the response variable?

In [3]:
df.plot(x='Average Monthly Temperature', y='Average Bill', kind='scatter')

From visual inspection, what is the relationship between the variables?

For shure not linear! Rather of degree two or higher

Fit a simple linear regression using OLS.  

In [4]:
import statsmodels.api as sms
import numpy as np
X = df[['Average Monthly Temperature']]
X = sms.add_constant(X)
#X['squared'] = df['Average Monthly Temperature'] **2
y = df[['Average Bill']]

# Create an OLS model
model = sms.OLS(y, X)
results = model.fit()
results.summary()

/Users/gabriel/anaconda3/envs/nf/lib/python3.6/site-packages/scipy/stats/stats.py:1394: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=12
  "anyway, n=%i" % int(n))


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:           Average Bill   R-squared:                       0.124
Model:                            OLS   Adj. R-squared:                  0.036
Method:                 Least Squares   F-statistic:                     1.416
Date:                Mon, 28 Oct 2019   Prob (F-statistic):              0.261
Time:                        13:16:17   Log-Likelihood:                -53.778
No. Observations:                  12   AIC:                             111.6
Df Residuals:                      10   BIC:                             112.5
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
===============================================================================================
                                  coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------------------
const                         143.6228     25.995      5.525      0.000      85.701     201.544
Average Monthly Temperature    -0.4799      0.403     -1.190      0.261      -1.378       0.419
==============================================================================
Omnibus:                        0.570   Durbin-Watson:                   1.176
Prob(Omnibus):                  0.752   Jarque-Bera (JB):                0.588
Skew:                           0.272   Prob(JB):                        0.745
Kurtosis:                       2.062   Cond. No.                         248.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Calculate and interpret the value of $R^2$.  

R-Value is with 0.124 really bad. Which supports the impression we had from the graphical interpretation. 

Interpret the intercept and slope coefficients in terms a home owner would understand.

Since the r2-value is really bad the model seems to fit the data not really good. The slopes says how the dependent variable changes in respect to the dependent variable.

What is the 95% confidence interval for the intercept?  

Is the average monthly temperature a significant predictor of the average utility bill? Why or why not?  

Use the least squares line to estimate the average utility bill if the average monthly temperature is 120 degrees. Do you think that your answer is reasonable? Why or why not? 

<br>

<details><summary>
Click here for a hint…
</summary>
model.predict(…)  
http://statsmodels.sourceforge.net/0.6.0/examples/notebooks/generated/predict.html
</details>

In [5]:
prediction = results.predict((1,120))

Since model fit is very poor. Its prediction could be take with caution.

<br>
<br> 
<br>

----